packages/protocol/test/governance/epochrewards.ts

```diff
@@ -0,0 +1,569 @@
+          resp = await epochRewards.setMaxValidatorEpochPayment(newPayment)
+        })
+
+        it('should set the max validator epoch payment', async () => {
+          assertEqualBN(await epochRewards.maxValidatorEpochPayment(), newPayment)
+        })
+
+        it('should emit the MaxValidatorEpochPaymentSet event', async () => {
+          assert.equal(resp.logs.length, 1)
+          const log = resp.logs[0]
+          assertContainSubset(log, {
+            event: 'MaxValidatorEpochPaymentSet',
+            args: {
+              payment: newPayment,
+            },
+          })
+        })
+
+        describe('when called by a non-owner', () => {
+          it('should revert', async () => {
+            await assertRevert(
+              epochRewards.setMaxValidatorEpochPayment(newPayment, {
+                from: nonOwner,
+              })
+            )
+          })
+        })
+      })
+
+      describe('when the payment is the same', () => {
+        it('should revert', async () => {
+          await assertRevert(epochRewards.setMaxValidatorEpochPayment(maxValidatorEpochPayment))
+        })
+      })
+    })
+  })
+
+  describe('#setRewardsMultiplierAdjustmentFactors()', () => {
+    describe('when the factors are different', () => {
+      const newFactors = {
+        underspend: rewardsMultiplierAdjustments.underspend.plus(1),
+        overspend: rewardsMultiplierAdjustments.overspend.plus(1),
+      }
+
+      describe('when called by the owner', () => {
+        let resp: any
+
+        beforeEach(async () => {
+          resp = await epochRewards.setRewardsMultiplierAdjustmentFactors(
+            newFactors.underspend,
+            newFactors.overspend
+          )
+        })
+
+        it('should set the new rewards multiplier adjustment factors', async () => {
+          const [underspend, overspend] = await epochRewards.getRewardsMultiplierAdjustmentFactors()
+          assertEqualBN(underspend, newFactors.underspend)
+          assertEqualBN(overspend, newFactors.overspend)
+        })
+
+        it('should emit the RewardsMultiplierAdjustmentFactorsSet event', async () => {
+          assert.equal(resp.logs.length, 1)
+          const log = resp.logs[0]
+          assertContainSubset(log, {
+            event: 'RewardsMultiplierAdjustmentFactorsSet',
+            args: {
+              underspend: newFactors.underspend,
+              overspend: newFactors.overspend,
+            },
+          })
+        })
+
+        describe('when called by a non-owner', () => {
+          it('should revert', async () => {
+            await assertRevert(
+              epochRewards.setRewardsMultiplierAdjustmentFactors(
+                newFactors.underspend,
+                newFactors.overspend,
+                {
+                  from: nonOwner,
+                }
+              )
+            )
+          })
+        })
+      })
+
+      describe('when the parameters are the same', () => {
```

Comment:
It may be worth testing that each factor can be adjusted separately (i.e. if one if the same and the other is different this call works.)

Reply:
Done